core/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -583,7 +575,7 @@ private void validate(Map<String, Settings> groupSettings) {
             ThreadPoolType correctThreadPoolType = THREAD_POOL_TYPES.get(key);
             // TODO: the type equality check can be removed after #3760/#6732 are addressed
             if (type != null && !correctThreadPoolType.getType().equals(type)) {
-                throw new IllegalArgumentException("setting " + THREADPOOL_GROUP + key + ".type to " + type + " is not permitted; must be " + correctThreadPoolType.getType());
+                throw new IllegalArgumentException("setting " + THREADPOOL_GROUP_SETTING.getKey() + key + ".type to " + type + " is not permitted; must be " + correctThreadPoolType.getType());
```

Comment:
Its all a blur now. So you intend for `getKey` to be required. Cool. The next time I look I'll remember that.